src/ngraph_encapsulate_op.cc

```diff
@@ -309,6 +309,60 @@ class NGraphEncapsulateOp : public OpKernel {
 
       auto function_size = ng_function->get_graph_size() / 1024;  // kb unit
 
+      // Get the serialized ops and stored the allreduce ops to a vector and
+      ng::NodeVector allreduce_op_list;
+      for (const shared_ptr<ng::Node>& node : ng_function->get_ordered_ops()) {
+        if (node->get_name().compare(0, 9, "AllReduce") == 0) {
+          allreduce_op_list.push_back(node);
+          // Diagnosis to check the input, dependencies, and output for each
+          // Allreduce op
+          if (std::getenv("NGRAPH_DISABLE_LOGGING") == nullptr) {
+#if defined NGRAPH_DISTRIBUTED
+            ngraph::Distributed dist;
+            int Rank_ID;
+            Rank_ID = dist.get_rank();
+            std::cout << "NGTF_Rank: " << Rank_ID << "  " << node->get_name()
+                      << "(";
+            vector<string> inputs;
+            for (const ng::descriptor::Input& input : node->get_inputs()) {
+              inputs.push_back(input.get_tensor().get_name());
```

Comment:
Interesting, Do we have tensor and tensor names at this stage?

Reply:
It seems yes. It will give the input tensor name for allreduce ops.